### PR TITLE
Document interaction between `networking.useNetworkd` and `systemd.networkd.enable`

### PR DESCRIPTION
**Pull request by asymmetric**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/system/boot/networkd.nix [hunk 1]
```diff
@@ -843,6 +843,8 @@ let
       type = types.bool;
       description = ''
         Whether to manage network configuration using <command>systemd-network</command>.
+
+        This also enables <option>systemd.networkd</option>.
```
Comment (dasJ) [inline nixos/modules/system/boot/networkd.nix:847]:
```suggestion
        This also sets <option>systemd.networkd.enable</option>.
```

**Review by dasJ — CHANGES_REQUESTED** [review]

**Comment by asymmetric** [comment]
I decided to do it like it's done in [other](https://github.com/NixOS/nixpkgs/tree/aea844573e114442b5f35acc209f4e10d67e0698/nixos/modules/services/backup/tsm.nix#L14) [places](https://github.com/NixOS/nixpkgs/tree/aea844573e114442b5f35acc209f4e10d67e0698/nixos/modules/programs/turbovnc.nix#L22)

File: nixos/modules/system/boot/networkd.nix [hunk 2]
```diff
@@ -928,6 +928,8 @@ let
       type = types.bool;
       description = lib.mdDoc ''
         Whether to manage network configuration using {command}`systemd-network`.
+        
+        This also enables <option>systemd.networkd.enable</option>.
```
Comment (ncfavier) [inline nixos/modules/system/boot/networkd.nix:932]:
```suggestion
        This also enables {option}`systemd.networkd.enable`.
```

**Review by ncfavier — CHANGES_REQUESTED** [review]

**Review by ncfavier — APPROVED** [review]

**Comment by ncfavier** [comment]
Please remove trailing whitespace

**Comment by ncfavier** [comment]
Please fix typos in the commit message

**Comment by asymmetric** [comment]
Eagled eyes! Done.

**Pull request merged**